zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -0,0 +1,651 @@
+            sid_column_ixs,
+            mask,
+            input_array,
+            missing_value,
+            is_missing_value[np.int64_t],
+            array_kind,
+        )
+    elif kind == 'M':
+        return _array_for_column_impl[np.int64_t, AsArrayKind](
+            dtype,
+            out_array.view('int64'),
+            size,
+            ts_ixs,
+            asof_ixs,
+            sids,
+            sid_column_ixs,
+            mask,
+            input_array.view('int64'),
+            missing_value.view('int64'),
+            is_missing_value[np.int64_t],
+            array_kind,
+        )
+    elif kind == 'f':
+        if isnan(missing_value):
+            return _array_for_column_impl[np.float64_t, AsArrayKind](
+                dtype,
+                out_array,
+                size,
+                ts_ixs,
+                asof_ixs,
+                sids,
+                sid_column_ixs,
+                mask,
+                input_array,
+                missing_value,
+                is_missing_nan,
+                array_kind,
+            )
+        else:
+            return _array_for_column_impl[np.float64_t, AsArrayKind](
+                dtype,
+                out_array,
+                size,
+                ts_ixs,
+                asof_ixs,
+                sids,
+                sid_column_ixs,
+                mask,
+                input_array,
+                missing_value,
+                is_missing_value[np.float64_t],
+                array_kind,
+            )
+    elif kind == 'O':
+        return _array_for_column_impl[object, AsArrayKind](
+            dtype,
+            out_array,
+            size,
+            ts_ixs,
+            asof_ixs,
+            sids,
+            sid_column_ixs,
+            mask,
+            input_array,
+            missing_value,
+            is_missing_value[object],
+            array_kind,
+        )
+    elif kind == 'b':
+        return _array_for_column_impl[np.uint8_t, AsArrayKind](
+            dtype,
+            out_array.view('uint8'),
+            size,
+            ts_ixs,
+            asof_ixs,
+            sids,
+            sid_column_ixs,
+            mask,
+            input_array.view('uint8'),
+            int(missing_value),
+            is_missing_value[np.uint8_t],
+            array_kind,
+        )
+    else:
+        raise TypeError('unknown column dtype: %r' % input_array.dtype)
+
+
+cdef arrays_from_rows(DatetimeIndex_t dates,
+                      object data_query_time,
+                      object data_query_tz,
+                      object assets,
+                      np.ndarray[np.int64_t] sids,
+                      object mask,
+                      list columns,
+                      object all_rows,
+                      AsArrayKind array_kind):
+    cdef dict column_ixs = dict(zip(assets, range(len(assets))))
+
+    cdef Py_ssize_t n
+    cdef list ts_dates_list
+    if data_query_time is not None:
```

Comment:
unrelated: why is this in a module called `trading_calendar` when it has nothing to do with the trading calendar.